src/corefx/System.Globalization.Native/calendarData.cpp

```diff
@@ -112,41 +123,117 @@ const char* GetCalendarName(CalendarId calendarId)
 	switch (calendarId)
 	{
 		case JAPAN:
-			return "japanese";
+			return JAPANESE_NAME;
 		case THAI:
-			return "buddhist";
+			return BUDDHIST_NAME;
 		case HEBREW:
-			return "hebrew";
-		case CHINESELUNISOLAR:
-		case KOREANLUNISOLAR:
-		case JAPANESELUNISOLAR:
-		case TAIWANLUNISOLAR:
-			return "chinese";
+			return HEBREW_NAME;
+		case KOREA:
+			return DANGI_NAME;
 		case PERSIAN:
-			return "persian";
+			return PERSIAN_NAME;
 		case HIJRI:
+			return ISLAMIC_NAME;
 		case UMALQURA:
-			return "islamic";
+			return ISLAMIC_UMALQURA_NAME;
+		case TAIWAN:
+			return ROC_NAME;
 		case GREGORIAN:
 		case GREGORIAN_US:
 		case GREGORIAN_ARABIC:
 		case GREGORIAN_ME_FRENCH:
 		case GREGORIAN_XLIT_ENGLISH:
 		case GREGORIAN_XLIT_FRENCH:
-		case KOREA:
 		case JULIAN:
 		case LUNAR_ETO_CHN:
 		case LUNAR_ETO_KOR:
 		case LUNAR_ETO_ROKUYOU:
 		case SAKA:
-		case TAIWAN:
+		// don't support the lunisolar calendars until we have a solid understanding
+		// of how they map to the ICU/CLDR calendars
+		case CHINESELUNISOLAR:
+		case KOREANLUNISOLAR:
+		case JAPANESELUNISOLAR:
+		case TAIWANLUNISOLAR:
 		default:
-			return "gregorian";
+			return GREGORIAN_NAME;
 	}
 }
 
 /*
 Function:
+GetCalendarId
+
+Gets the associated CalendarId for the ICU calendar name.
+*/
+CalendarId GetCalendarId(const char* calendarName)
+{
+	if (strcasecmp(calendarName, GREGORIAN_NAME) == 0)
+		//TODO: what about the other gregorian types?
+		return GREGORIAN;
+	else if (strcasecmp(calendarName, JAPANESE_NAME) == 0)
+		return JAPAN;
+	else if (strcasecmp(calendarName, BUDDHIST_NAME) == 0)
+		return THAI;
+	else if (strcasecmp(calendarName, HEBREW_NAME) == 0)
+		return HEBREW;
+	else if (strcasecmp(calendarName, DANGI_NAME) == 0)
+		return KOREA;
+	else if (strcasecmp(calendarName, PERSIAN_NAME) == 0)
+		return PERSIAN;
+	else if (strcasecmp(calendarName, ISLAMIC_NAME) == 0)
+		return HIJRI;
+	else if (strcasecmp(calendarName, ISLAMIC_UMALQURA_NAME) == 0)
+		return UMALQURA;
+	else if (strcasecmp(calendarName, ROC_NAME) == 0)
+		return TAIWAN;
```

Comment:
Because the value the constant represents is "roc".  I think it makes for easier reading code if the constant's value and name are close/similar.  That way people can tell the value without having to go find the definition.
This is the same naming pattern used for "dangi" being DANGI_NAME instead of KOREA, "buddhist" being BUDDHIST_NAME instead of THAI, etc.